src/Microsoft.TestPlatform.Utilities/CodeCoverageRunSettingsProcessor.cs

```diff
@@ -0,0 +1,515 @@
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+    using System.Xml;
+    using System.Xml.XPath;
+
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+
+    /// <summary>
+    /// Represents the run settings processor for code coverage data collectors.
+    /// </summary>
+    public class CodeCoverageRunSettingsProcessor
+    {
+        #region Type Members
+        /// <summary>
+        /// Represents the exclusion type for code coverage run settings.
+        /// </summary>
+        private class Exclusion
+        {
+            /// <summary>
+            /// Represents the <see cref="XPathNavigator"/> style path of the exclusion type.
+            /// </summary>
+            private string path;
+
+            /// <summary>
+            /// Gets the path components for the current exclusion type.
+            /// </summary>
+            public IEnumerable<string> PathComponents { get; }
+
+            /// <summary>
+            /// Gets the exclusion rules for the current exclusion type.
+            /// </summary>
+            public IDictionary<string, XmlNode> ExclusionRules { get; }
+
+            /// <summary>
+            /// Gets the actual exclusion type path generated from the individual path components.
+            /// </summary>
+            public string Path
+            {
+                get
+                {
+                    if (string.IsNullOrEmpty(this.path))
+                    {
+                        this.path = Exclusion.BuildPath(this.PathComponents);
+                    }
+
+                    return this.path;
+                }
+            }
+
+            /// <summary>
+            /// Constructs an <see cref="Exclusion"/> object.
+            /// </summary>
+            /// 
+            /// <param name="pathComponents">The path split in components.</param>
+            public Exclusion(IEnumerable<string> pathComponents)
+            {
+                this.path = string.Empty;
+                this.PathComponents = pathComponents;
+                this.ExclusionRules = new Dictionary<string, XmlNode>();
+            }
+
+            /// <summary>
+            /// Assembles a relative path from the path given as components.
+            /// </summary>
+            /// 
+            /// <returns>A relative path built from path components.</returns>
+            public static string BuildPath(IEnumerable<string> pathComponents)
+            {
+                var path = ".";
+
+                foreach (var component in pathComponents)
+                {
+                    path += "/" + component;
+                }
+
+                return path;
+            }
+        }
+        #endregion
+
+        #region Members
+        #region Default Settings String
+        /// <summary>
+        /// Represents the default settings for the code coverage data collector.
+        /// </summary>
+        private static readonly string DefaultSettings = string.Join(
```

Comment:
As you initialize variable it is better to change it to "const"

Reply:
The const specifier doesn't work in this context and causes a compilation error (see [this](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/const)).